flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraph.java

```diff
@@ -148,6 +162,33 @@ public Logger getLogger() {
         return logger;
     }
 
+    protected Throwable extractError(TaskExecutionStateTransition taskExecutionStateTransition) {
+        Throwable cause = taskExecutionStateTransition.getError(userCodeClassLoader);
+        if (cause == null) {
+            cause = new FlinkException("Unknown failure cause. Probably related to FLINK-21376.");
+        }
+        return cause;
+    }
+
+    protected Optional<ExecutionVertexID> extractExecutionVertexID(
+            TaskExecutionStateTransition taskExecutionStateTransition) {
+        return executionGraph.getExecutionVertexId(taskExecutionStateTransition.getID());
+    }
+
+    protected static Optional<RootExceptionHistoryEntry> convertFailures(
+            Function<ExecutionVertexID, Optional<ExecutionVertex>> lookup,
```

Comment:
This has a weird signature, why can't we simply pass an execution graph here?

Reply:
The codestyle also forbids optional method parameters.

Reply:
the method no longer exists

Reply:
There is a similar method in the `Failure` class. I guess giving the parameter a more descriptive name would do the trick

```
    /**
     * Convert the failure into an {@link ExceptionHistoryEntry}.
     *
     * @param findExecutionFn Function that looks up additional information on the {@link
     *     ExecutionVertexID}, which can be used for constructing the {@link ExceptionHistoryEntry}.
     * @return The {@link ExceptionHistoryEntry} derived for this failure.
     */
    public abstract ExceptionHistoryEntry toExceptionHistoryEntry(
            Function<ExecutionVertexID, Execution> findExecutionFn);
```

One additional thought about naming getters, it's good to distinguish between getters that return an actual value (`getXXX`) and getters that return an optional (`findXXX`).